### DISCUSSION TITLE
Make optional peer dependencies way more useful

### DISCUSSION THREAD

**Discussion by Roaders** [npm/feedback#413] [CLI] [answered]
I was pretty pleased when I saw that there was a way to make peer dependencies optional. I have quite a few places where this will be useful - except it seems that optional peer dependencies do not work the way that I expected. Instead it seems to basically turn off the peer dependencies completely (or the warning at least).

I have a project that among other things provides sample configs for eslint. Some of the features that are used in these config files require a certain version of eslint so I specify eslint as a peer dependency with the correct version. However, this package provides lots of other config as well and some consumers won't use eslint so they wont install it.

In my opinion optional peer dependencies should allow the peer not to be installed, but if it IS installed throw an error if an incorrect version is installed.

**Comment by MylesBorins** [comment, marked as answer]
@Roaders if you have an explicit way in which you think the resolution or reification algorithm should be changed you should open an issue over in https://github.com/npm/rfcs to kick off a discussion with the npm CLI team

**Comment by Roaders** [comment]
added RFC:

https://github.com/npm/rfcs/pull/386
